src/native/fs/fs_napi.cpp

```diff
@@ -480,20 +544,31 @@ struct FileOpen : public FSWorker
 {
     std::string _path;
     int _fd;
-    FileOpen(const Napi::CallbackInfo& info) : FSWorker(info), _fd(0)
+    int _flags;
+    mode_t _mode;
+    FileOpen(const Napi::CallbackInfo& info) 
+        : FSWorker(info)
+        , _fd(0)
+        , _flags(0)
+        , _mode(0666)
     {
         _path = info[1].As<Napi::String>();
-        // TODO - info[1] { mode, readonly }
+        if (info.Length() > 2 && !info[2].IsUndefined()) {
+            _flags = _stringToFlags(info[2].As<Napi::String>());
+        }
+        if (info.Length() > 3 && !info[3].IsUndefined()) {
+            _mode = unsigned(info[3].As<Napi::Number>());
```

Comment:
I think you should use `info[3].As<Napi::Number>().Uint32Value()`

https://github.com/nodejs/node-addon-api/blob/main/doc/number.md#uint32value